cpp/src/io/orc/reader_impl.cu

```diff
@@ -177,15 +177,23 @@ class metadata {
         selection.emplace_back(&ff.stripes[stripe_idx], nullptr);
         stripe_rows += ff.stripes[stripe_idx].numberOfRows;
       }
+      // row_start is 0 if stripes are set. If this is not true anymore, then
+      // row_start needs to be subtracted to get the correct row_count
+      CUDF_EXPECTS(row_start == 0, "Start row index should be 0");
       row_count = static_cast<size_type>(stripe_rows);
     } else {
       row_start = std::max(row_start, 0);
       if (row_count < 0) {
```

Comment:
I don't see a test coverage where `row_count < 0`, can you please point me to it.

Reply:
`row_count < 0` is true whenever `orc_reader_options::builder` does not call its `num_rows()` function. In that case `orc_reader_options::_num_rows` is left at its initial value of `-1`.

In this PR that case is covered by `SkipRowTest.test(int skip_rows, int file_num_rows)` called as
```
skip_row.test(50, 75);
skip_row.test(2, 100);
```
in the `OrcReaderTest::CombinedSkipRowTest`